FogCreek/Glitch-Community
# various dart: completes account closing work

## Links
* https://various-dart.glitch.me
* https://app.clubhouse.io/glitch/story/4033/update-frontend-to-use-new-updated-endpoints
* https://docs.google.com/document/d/1ya5WQraU0OPXf22uCsTGhaNnFU-3XyVLHSqGHzMkTPk/edit#heading=h.tcauhdj77y1d

## Changes:
* Hooks modal up to backend
* Sends email to complete deletion when you have transferred all projects/teams
* On clicking link in email, checks token validity and closes account
* Logs you out on account close
* Changes wording from "delete" to "close" (since not everything is deleted on our end)
* If you don't have projects/teams to transfer, skip those dialogs

## How To Test:
* Sign into various-dart
* Turn on Account Deletion switch
* Go to `/settings`, walk through the steps to close your account (use a test account! make sure it has some projects and/or teams with other users, where your account is the admin)
* Get an email, click the link
* You'll need to change the link to be various-dart.glitch.me/delete?token=[whatever]
* Check to make sure your account is gone, and that you've been logged out
* You can also try /delete?token=[garbage] when signed into an account and make sure you get an error

## Feedback I'm looking for:

## Things left to do before deploying:
- [x] Have support write articles explaining ownership transfer and other account-closing-specifics (ch5319, ch5320, ch5321, ch5322, ch5323) 
- [x] Fill in those links
- [ ] Take out feature switch? At least have a plan as to when to do so.
- [x] Is there a way to confirm that the feedback form is working and going where folks want it?
- [x] Make sure confirmation email is properly styled (ch6301)
- [x] Add analytics (ch3242)
